src/auth/auth.service.ts

```diff
@@ -46,6 +46,14 @@ export class AuthService {
 
   async register(registerDto: RegisterDto): Promise<RegisterResponseDto> {
     const hash = await argon2.hash(registerDto.password);
+    const existingUser = await this.userModel.findOne({
+      username: registerDto.username,
+    });
+    if (existingUser) {
+      throw new UnauthorizedException({
+        message: 'Username is already taken',
+      });
+    }
```

Comment:
When you try to create a user doc that duplicate with username in database, mongoose will throw a error (Code:E11000) Please check from this. [May be use try-catch or `Promise.prototype.catch(...)` function]
[Ref](https://stackoverflow.com/questions/24430220/e11000-duplicate-key-error-index-in-mongodb-mongoose)